lib/sequel/extensions/sql_comments.rb

```diff
@@ -71,7 +71,12 @@ def comment(comment)
           # SQL is appened to the query after the comment is added,
           # it will become part of the comment unless it is preceded
           # by a newline.
-          sql << comment
+          if sql.frozen?
+            sql += comment
+            sql.freeze
+          else
+            sql << comment
```

Comment:
This `else` branch may be redundant if `sql` is always guaranteed to be frozen, but IDK. The intent here is to make sure that frozen strings remain frozen after a comment is added, and that not-frozen strings do not get inadvertently frozen after a comment is added.